designer/server/src/main/scala/pl/touk/nussknacker/ui/services/MigrationApiHttpService.scala

```diff
@@ -0,0 +1,174 @@
+package pl.touk.nussknacker.ui.services
+
+import cats.data.EitherT
+import com.typesafe.config.Config
+import com.typesafe.scalalogging.LazyLogging
+import pl.touk.nussknacker.engine.api.process.{ProcessIdWithName, ProcessName}
+import pl.touk.nussknacker.engine.util.Implicits._
+import pl.touk.nussknacker.restmodel.scenariodetails.{ScenarioParameters, ScenarioWithDetails}
+import pl.touk.nussknacker.restmodel.validation.ValidationResults.ValidationErrors
+import pl.touk.nussknacker.security.Permission
+import pl.touk.nussknacker.ui.{NuDesignerError, UnauthorizedError}
+import pl.touk.nussknacker.ui.api.{AuthorizeProcess, ListenerApiUser, MigrationApiEndpoints}
+import pl.touk.nussknacker.ui.listener.ProcessChangeEvent.OnSaved
+import pl.touk.nussknacker.ui.listener.{ProcessChangeEvent, ProcessChangeListener, User}
+import pl.touk.nussknacker.ui.process.ProcessService
+import pl.touk.nussknacker.ui.process.ProcessService.{
+  CreateScenarioCommand,
+  FetchScenarioGraph,
+  GetScenarioWithDetailsOptions,
+  UpdateScenarioCommand
+}
+import pl.touk.nussknacker.ui.process.migrate.{MigrationToArchivedError, MigrationValidationError}
+import pl.touk.nussknacker.ui.process.processingtype.ProcessingTypeDataProvider
+import pl.touk.nussknacker.ui.process.repository.ProcessDBQueryRepository.ProcessNotFoundError
+import pl.touk.nussknacker.ui.process.repository.ProcessRepository.RemoteUserName
+import pl.touk.nussknacker.ui.process.repository.UpdateProcessComment
+import pl.touk.nussknacker.ui.security.api.{AuthenticationResources, LoggedUser}
+import pl.touk.nussknacker.ui.uiresolving.UIProcessResolver
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+class MigrationApiHttpService(
+    config: Config,
+    authenticator: AuthenticationResources,
+    processService: ProcessService,
+    processResolver: ProcessingTypeDataProvider[UIProcessResolver, _],
+    processAuthorizer: AuthorizeProcess,
+    processChangeListener: ProcessChangeListener,
+    useLegacyCreateScenarioApi: Boolean
+)(implicit val ec: ExecutionContext)
+    extends BaseHttpService(config, authenticator)
+    with LazyLogging {
+
+  private val remoteEnvironmentApiEndpoints = new MigrationApiEndpoints(authenticator.authenticationMethod())
+  private val passUsernameInMigration       = true
+
+  expose {
+    remoteEnvironmentApiEndpoints.migrateEndpoint
+      .serverSecurityLogic(authorizeKnownUser[NuDesignerError])
+      .serverLogicEitherT { implicit loggedUser => migrateScenarioRequest =>
+        {
+          val scenarioWithDetailsForMigrations = migrateScenarioRequest.scenarioWithDetailsForMigrations
```

Comment:
1. Let's extract a domain service for that instead of keeping everything in the infrastructure layer.
2. I see that you added a lot of code but didn't remove any. What's going one? Can you provide some wider context on what are you planning to do in the PR description?

Reply:
I basically rewritten the content of `pl.touk.nussknacker.ui.process.migrate` `migrate` function. In this way we don't rely on four endpoints.

Reply:
So what do you planning with the old one? What about tests?